django/core/checks/security/base.py

```diff
@@ -235,6 +239,65 @@ def check_referrer_policy(app_configs, **kwargs):
     return []
 
 
+@register(Tags.security, deploy=True)
+def check_cache_and_media_root(app_configs, **kwargs):
+    if not settings.MEDIA_ROOT:
+        return []
+    _media_root = os.path.abspath(settings.MEDIA_ROOT)
+    for alias in settings.CACHES.keys():
+        if isinstance(caches[alias], FileBasedCache):
+            location_check = (
+                _media_root,
+                caches[alias]._dir,
+            )
+            if os.path.commonpath(location_check) == _media_root:
+                return [Warning(
+                    "Your cache configuration might expose or corrupt your file system.",
+                    hint=f"Your cache '{alias}' is inside your 'MEDIA_ROOT'",
```

Comment:
I think this might read better as:
```suggestion
                    hint=f"Your '{alias}' cache is inside your {name}.",
```